Tests/MessagingPush/UserNotificationsFramework/UserNotificationsFrameworkAdapterTest.swift

```diff
@@ -0,0 +1,32 @@
+@testable import CioMessagingPush
+import Foundation
+import SharedTests
+import XCTest
+
+class UserNotificationsFrameworkAdapterTest: IntegrationTest {
+    // MARK: UNUserNotificationCenterDelegate
+
+    /**
+     The SDK comes with a `UNUserNotificationCenterDelegate` instance that is provided to the OS. The OS keeps a reference to it and will call it to handle push notification events, such as when a push is clicked.
+
+     It's very important that the object that the OS has in-memory, stays in-memory otherwise the SDK will not be called to handle push events.
+
+     This test makes sure that the `UNUserNotificationCenterDelegate` instance stays in-memory, even after other SDK dependencies may be destroyed.
+     */
+    func test_expectSdkUNUserNotificationCenterDelegateInMemoryAfterSdkReinitialized() {
```

Comment:
New test that makes sure that the `UNUserNotificationCenterDelegate` delegate stays in-memory.